openedx/core/djangoapps/discussions/api/providers.py

```diff
@@ -8,7 +8,7 @@
 from ..models import DiscussionProviderConfig
 
 
-def get_discussion_provider(context_key: LearningContextKey) -> Optional[Type[DiscussionApp]]:
+def get_discussion_provider(context_key: LearningContextKey) -> 'Optional[Type[DiscussionApp]]':
```

Comment:
This was causing a lot of issues. It seems there is a bug in Python typing with Python 3.5 versions below 3.5.3
https://github.com/python/typing/issues/266